app/src/main/java/com/keylesspalace/tusky/util/LinkHelper.java

```diff
@@ -117,20 +117,74 @@ public void onClick(View widget) {
             /* Add zero-width space after links in end of line to fix its too large hitbox.
              * See also : https://github.com/tuskyapp/Tusky/issues/846
              *            https://github.com/tuskyapp/Tusky/pull/916 */
-            if(end >= builder.length()){
+            if (end >= builder.length()){
                 builder.insert(end, "\u200B");
             } else {
                 if(builder.subSequence(end, end + 1).toString().equals("\n")){
                     builder.insert(end, "\u200B");
                 }
             }
-
+            // if (end >= builder.length() ||
+            //         builder.subSequence(end, end + 1).toString().equals("\n")){
+            //     builder.insert(end, "\u200B");
+            // }
         }
+
         view.setText(builder);
         view.setLinksClickable(true);
         view.setMovementMethod(LinkMovementMethod.getInstance());
     }
 
+    /**
+     * Make a text listing mentions and makes them clickable, associating them with
+     * callbacks to notify when they're clicked.
+     *
+     * @param mentions any '@' mentions which are known to be in the content
+     * @param listener to notify about particular spans that are clicked
+     */
+    public static @Nullable CharSequence makeMentionsText(
+            @Nullable Status.Mention[] mentions, final LinkListener listener) {
+        if (mentions == null || mentions.length == 0) {
+            return null;
+        }
+        SpannableStringBuilder builder = new SpannableStringBuilder();
+        int start = 0;
+        int end = 0;
+        int flags;
+        boolean firstMention = true;
+        for (Status.Mention mention : mentions) {
+            String accountUsername = mention.getLocalUsername();
+            final String accountId = mention.getId();
+            ClickableSpan customSpan = new ClickableSpanNoUnderline() {
+                @Override
+                public void onClick(View widget) { listener.onViewAccount(accountId); }
+            };
+
+            customSpan = new CustomURLSpan(mention.getUrl()) {
```

Comment:
The `ClickableSpan` is created twice, only the second one is used. I believe the first one is the one we want, as it opens profiles directly and not via url.